Makefile

```diff
@@ -77,14 +77,26 @@ clean: ## Delete build output
 
 ##@ Testing
 
-.PHONY: test
-test: generate ## Run all unit tests
+.PHONY: test-unit
+test-unit:
 	@go test -race -covermode=atomic -coverprofile=coverage-unit.out -timeout 500ms -tags=unit ./...
+
+.PHONY: test-integration
+test-integration:
 	@go test -race -covermode=atomic -coverprofile=coverage-integration.out -timeout 15s -tags=integration ./...
+
+.PHONY: test-generative
+test-generative:
 # Given the nature of generative tests the test timeout is increased from 500ms
 # to 30s to accommodate many samples being generated and test cases being run.
 	@go test -race -covermode=atomic -coverprofile=coverage-generative.out -timeout 30s -tags=generative ./...
 
+.PHONY: test
+test: generate ## Run all unit tests
+	@$(MAKE) test-unit
+	@$(MAKE) test-integration | grep -v '\[no test files\]'
+	@$(MAKE) test-generative | grep -v '\[no test files\]'
```

Comment:
Why not?

```suggestion
test: generate test-unit test-integration test-generative
```

Reply:
See commit message where I answer this question.

Reply:
It's because they each run in parallel. There might be another way to make them run sequentially though..

Reply:
Saves us time to run them in parallel, perhaps we could tweak the way results were outputted, though not sure we loose that much on performance...